### ISSUE TITLE
wpa_supplicant is not correctly shutdown/killed

### ISSUE THREAD

**Issue by patgrosse** [intrig-unicamp/mininet-wifi#25]
From [wifiModule.py line 58](https://github.com/intrig-unicamp/mininet-wifi/blob/master/mininet/wifiModule.py#L58):

```python
        try:
            h = subprocess.check_output("ps -aux | grep -ic \'wpa_supplicant -B -Dnl80211\'",
                                                          shell=True)
            if h >= 2:
                os.system('pkill -f \'wpa_supplicant -B -Dnl80211\'')
        except Exception as e:
            pass # e => Command 'ps -aux | grep -ic 'wpa_supplicant -B -Dnl80211'' returned non-zero exit status 1
```
grep exit status 1 means no lines have been found. But processes are running, even after net is stopped.

Used example: wifiDirect.py

**Comment by ramonfontes** [comment]
Have you changed wpa_supplicant in order to enable wifi-direct?

**Comment by patgrosse** [comment]
In the CONFIG_P2P in defconfig? Yes, I did

**Comment by ramonfontes** [comment]
The processes aren't running here. 

**After starting wifiDirecty.py:**
ps -aux | grep wpa
alpha      7775  0.0  0.0  44348  7112 ?        Ss   18:05   0:00 /sbin/wpa_supplicant -u -s -O /run/wpa_supplicant
alpha     32258  0.0  0.0  26492   536 ?        Ss   20:27   0:00 wpa_supplicant -B -Dnl80211 -ista1-wlan0 -d -csta1_wifiDirect.conf
alpha     32263  0.0  0.0  26492   536 ?        Ss   20:27   0:00 wpa_supplicant -B -Dnl80211 -ista2-wlan0 -d -csta2_wifiDirect.conf
alpha     32282  0.0  0.0  14236   988 pts/4    S+   20:28   0:00 grep --color=auto wpa

**After net.stop():**
ps -aux | grep wpa
alpha      7775  0.0  0.0  44348  7112 ?        Ss   18:05   0:00 /sbin/wpa_supplicant -u -s -O /run/wpa_supplicant
alpha     32331  0.0  0.0  14236   936 pts/4    S+   20:28   0:00 grep --color=auto wpa

**Comment by patgrosse** [comment]
What I changed in defconfig (also copied it to .config, like it is done in util/install.sh):
![image](https://cloud.githubusercontent.com/assets/23578938/21766576/9e95ebf2-d66f-11e6-9ab4-aa3b8cfc3081.png)

The running wpa_supplicant processes (after the net is started and still active after net is stopped):
![image](https://cloud.githubusercontent.com/assets/23578938/21766588/af8a5b6e-d66f-11e6-9d57-b247978d67ff.png)

"Function test" with wifiDirect.py:
![image](https://cloud.githubusercontent.com/assets/23578938/21766605/c86de268-d66f-11e6-8700-e1ec8e549ec9.png)

**Comment by ramonfontes** [comment]
I think there is some difference between master's and multiple_instances' branch. Did you try it with master's branch and did you get the same behavior? Ping shouldn't work, cause we have to manually configure the interfaces for wifi-direct

**Comment by patgrosse** [comment]
This is on master ;) I did a full recompile (`make clean` in `wpa_supplicant`) and build Mininet-WiFi from master branch.

Okay if ping does not work by default, this could mean wifi direct is correctly configured, but the processes stay active (on my machine) after Mininet has finished.

**Comment by ramonfontes** [comment]
This is weird. I'll keep this issue open until observe the same behavior as you described.

**Comment by patgrosse** [comment]
Fixed in commit 88e14fb
